javascripts/govuk/anchor-buttons.js

```diff
@@ -0,0 +1,55 @@
+(function(global, settings) {
+  "use strict"; 
+       
+  var $ = global.jQuery;
+  var GOVUK = global.GOVUK || {};  
+
+  GOVUK.anchorButtons = {
+    
+    // default configuration that can be overridden by passing object as second parameter to module
+    config: $.extend({
+      // setting to true will disable the init function upon execution
+      disableInit: false,
+      // the target element(s) to attach the shim event to
+      selector: 'a[role="button"]',
+      // array of keys to match against upon the keyup event
+      keycodes: [
+        32 // spacekey
+      ],
+    }, settings),
+    
+    // event behaviour (not a typical anonymous function for resuse if needed)
+    triggerClickOnTarget: function triggerClickOnTarget(event) {
+      var code = event.charCode || event.keyCode; 
+      // if the keyCode/charCode from this event is in the keycodes array then 
+      if ($.inArray(code, this.config.keycodes) !== -1) {
+        event.preventDefault();
+        // trigger the target's click event
+        $(event.target).trigger("click");
+      }
+    },
+    
+    // init function (so it can be executed again if needed)
+    init: function init() { 
+      var $elms = $(this.config.selector);
```

Comment:
Could we attach a listener to the document rather than one for each button?

```js
$(document).on('keyup', this.config.selector, this.triggerClickOnTarget.bind(this));
```

My thought is that this way we only have one delegated listener rather than potentially a few?